tuareg-intern.el

```diff
@@ -0,0 +1,119 @@
+;;; tuareg-intern.el --- Helper functions
```

Comment:
You put too many things in a single PR.  There are tests above, now this is a refactoring of the library.  Please split your PR in small easily reviewable chunks.